data/modules/FlightLog/FlightLogEntries.lua

```diff
@@ -0,0 +1,397 @@
+	return { { "ERROR", "This should never be seen" } }
+end
+
+---@param entry string A user provided description of the event.
+---If non nil/empty this will cause the entry to be considered custom and not automatically deleted
+---@return nil
+function FlightLogEntry.Base:UpdateEntry( entry )
+
+	if self:IsCustom() then
+		FlightLogEntry.TotalDefaultElements = FlightLogEntry.TotalDefaultElements-1
+	end
+
+	if entry and #entry == 0 then entry = nil end
+	self.entry = entry
+
+	if self:IsCustom() then
+		FlightLogEntry.TotalDefaultElements = FlightLogEntry.TotalDefaultElements+1
+	end
+
+end
+
+---@param sort_date number The date to sort by (from epoch)
+---@param entry string|nil The user entered custom test for this entry
+---@param always_custom boolean Is this always treated as a custom entry
+function FlightLogEntry.Base:Constructor( sort_date, entry, always_custom )
+	-- a date that can be used to sort entries on TODO: remove this
+	self.sort_date = sort_date
+	-- the entry text associated with this log entry
+	if entry and #entry == 0 then entry = nil end
+	self.entry = entry
+	self.always_custom = always_custom
+	if self:IsCustom() then
+		FlightLogEntry.TotalDefaultElements = FlightLogEntry.TotalDefaultElements+1
+	end
+end
+
+
+--- convenience helper function
+--- Sometimes date is empty, e.g. departure date prior to departure
+--- TODO: maybe not return this at all then!
+---
+---@param date number The date since the epoch
+---
+---@return string  The date formatted
+function FlightLogEntry.Base.formatDate(date)
+	return date and Format.Date(date) or nil
+end
+
+--- Based on flight state, compose a reasonable string for location
+--- TODO: consider a class to represent, construct, store and format this
+---@param location string[] Array of string info, the first one is the  
+---@return string The formatted composite location.
+function FlightLogEntry.Base.composeLocationString(location)
+	return string.interp(l["FLIGHTLOG_"..location[1]],
+		{ primary_info = location[2],
+			secondary_info = location[3] or "",
+			tertiary_info = location[4] or "",})
+end
+
+---@class CurrentStatusLogEngtry : FlightLogEntry.Base
+--- Does not have any members, it grabs the current status live whenever requested
+FlightLogEntry.CurrentStatus = utils.class("FlightLogEntry.CurrentStatus", FlightLogEntry.Base )
+
+---@return boolean true if this is considered to have an entry
+function FlightLogEntry.CurrentStatus:CanHaveEntry()
+	return false
+end
+
+function FlightLogEntry.CurrentStatus:Constructor()
+	FlightLogEntry.Base.Constructor( self, Game.time, nil, true )
+end
+
+---@return string The name for this log entry type
+function FlightLogEntry.CurrentStatus:GetLocalizedName()
+	return l.PERSONAL_INFORMATION;
+end
+
+---@param earliest_first boolean set to true if your sort order is to show the earlist first dates
+---@return table<string, string>[] An array of key value pairs, the key being localized and the value being formatted appropriately.
```

Comment:
Note: `table<string, string>` is an annotation for a key-value table. The type annotation for this would be `{ [1]: string, [2]: string }[]`. (This should be applied to all instances of GetDataPairs.)